mfr/extensions/tabular/libs/h5py_scipy_tools.py

```diff
@@ -0,0 +1,60 @@
+import h5py
+import scipy.io
+
+from ..utilities import data_population, header_population
+from collections import OrderedDict
+
+
+def mat_v73(fp):
+    """Read and convert a mat v7.3+ file to JSON format using h5py
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+
+    workbook = h5py.File(fp.name, 'r')
+    sheets = OrderedDict()
+    variables = workbook.items()
+
+    for var in variables:
+        name = var[0]
+        data = var[1]
+        if type(data) is h5py.Dataset:
+            #h5py seems to swap cols/rows for some reason. this fixes it so it displays like it does in matlab
+            list_data = list(zip(*data.value.tolist()))
+            #add a default header to fix some data formats
+            header = [i for i in range(1, 1 + len(list_data[0]))]
+            values = data_population(list_data, headers=header)
+            header = header_population(header)
+            sheets[str(name)] = (header, values)
+
+    return sheets
+
+
+def mat_v7(fp):
+    """Read and convert a mat v7.0 and below file to JSON format using scipy.io.matload
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+
+    workbook = scipy.io.loadmat(fp.name)
+    sheets = OrderedDict()
+
+    for key in ('__header__', '__version__', '__globals__'):
+        workbook.pop(key, None)
+
+    for key in workbook:
+        list_data = workbook[key].tolist()
+        # add default header to fix rendering of some data formats
+        header = [i for i in range(1, 1 + len(list_data[0]))]
+        values = data_population(list_data, headers=header)
+        header = header_population(header)
+        sheets[str(key)] = (header, values)
+
+    return sheets
+
+def mat_h5py_scipy(fp):
```

Comment:
1. If an exception is expected, we can to document it using `:raises`.
2. For `try ... except ...`, what is the exact exception(s) that a `v7` parser throws on a `v73` file? We should be specific on exceptions.
3. **OPTIONAL** Is there alternative ways to identify the version other than exceptions?

Reply:
How about malformed/corrupted files? I guess that some exceptions are thrown. We should catch and handle them in `mat_v7()` or `mat_v73()` instead of throwing `500`. The front end can inform the user that the file cannot be rendered due to corrupted file or unsupported version.

Please note that this may applies for all types of files. We may need another clean-up ticket if we don't already handle corrupted files as a standard, which is out of the scope of this ticket. However, it is better to have this check for new ones we are adding. Not a blocker though.